packages/eas-cli/src/commandUtils/easCommand.ts

```diff
@@ -0,0 +1,37 @@
+import { Command } from '@oclif/command';
+
+import { AnalyticsEvent, flushAsync, initAsync, logEvent } from '../analytics';
+import { getUserAsync } from '../user/User';
+import { ensureLoggedInAsync } from '../user/actions';
+
+export default abstract class EasCommand extends Command {
+  /**
+   * When user data is unavailable locally, determines if the command will
+   * request it from the backend
+   * @returns boolean - if authentication is required
+   */
+  requiresAuthentication(): boolean {
```

Comment:
I think it'd be better to transform it to a member variable. I don't see any benefits from having a function for this. 

https://www.typescriptlang.org/play?#code/IYIwzgLgTsDGEAJYBthjAgggg3gKAQQAcoB7CAU3goBMEAlAUQEUBVASSYGUB9TVgCoAJRgDkB7AMKYJAeVEIAvAmgBXCgG48AXzx4UaDACEEFAB6UAdjQzZ8hAGalSACgCUuAoSSlLYUsgUAHTIpADmLhAAFgCWYEFMbJyMvPzCYhLScqJuWoS6uvqo6AiSphYU1raehCTkVJR0iRzcfIIi4lIy7PJKCA7AyGCaXk6uHvbesL7+gSHhkbHxzcmp7Rld2bleBXoulhQA7ghG7m5BY+5a+0elZxfOV0A

Reply:
I initially made it abstract to force devs to pay attention to it if they're making new commands. I ended up refactoring to the 'virtual' implementation you see here, but like you said, it's just simpler to use a member variable in this case.